build/Program.cs

```diff
@@ -0,0 +1,331 @@
+﻿using Microsoft.DotNet.ProjectModel;
+using System;
+using System.IO;
+using System.Linq;
+using NuGet.Versioning;
+using System.Collections.Generic;
+using LibGit2Sharp;
+using Newtonsoft.Json;
+using System.Text;
+
+namespace ConsoleApplication
+{
+    public class Program
+    {
+         const string fallbackTag = "CI";
+
+        public static void Main(string[] args)
+        {
+            // TODO add options to updating the version number for indirvidual projects
+
+            var resetmode = args.Contains("reset");
+
+            // find the project root where glbal.json lives
+            var root = ProjectRootResolver.ResolveRootDirectory(".");
+            //lets find the repo
+            var repo = new LibGit2Sharp.Repository(root);
+
+            //lets find all the project.json files in the src folder (don't care about versioning `tests`)
+            var projectFiles = Directory.EnumerateFiles(Path.Combine(root, "src"), Project.FileName, SearchOption.AllDirectories);
+
+            //open them and convert them to source projects
+            var projects = projectFiles.Select(x => ProjectReader.GetProject(x))
+                            .Select(x => new SourceProject(x, repo.Info.WorkingDirectory))
+                            .ToList();
+
+            if (resetmode)
+            {
+                ResetProject(projects);
+            }
+            else
+            {
+                CaclulateProjectVersionNumber(projects, repo);
+
+                UpdateVersionNumbers(projects);
+
+                CreateBuildScript(projects);
+                
+                foreach (var p in projects)
+                {
+                    Console.WriteLine($"{p.Name} {p.FinalVersionNumber}");
+                }
+            }
+        }
+
+        private static void CreateBuildScript(IEnumerable<SourceProject> projects)
+        {
+            var sb = new StringBuilder();
+            foreach (var p in projects)
+            {
+
+                sb.AppendLine($@"dotnet pack --configuration Release --output ""artifacts\bin\ImageSharp"" ""{p.ProjectFilePath}""");
+            }
+            File.Copy("build-inner.cmd", "build-inner.bak", true);
```

Comment:
it got populated by the build project after it discovered all the things to build... i've figured out a cleaner way for it to do it so this can now be dropped